src/surface/embedded_geometry_interface.cpp

```diff
@@ -278,11 +288,11 @@ void EmbeddedGeometryInterface::computeCornerAngles() {
     Vector3 pA = vertexPositions[he.vertex()];
     he = he.next();
     Vector3 pB = vertexPositions[he.vertex()];
-    he = he.next();
+    do {
```

Comment:
This function has quadratic complexity in the face degree, when it doesn't need to. It could be implemented with linear complexity by working one face at a time and maintaining a leading a trailing reference while orbiting the face.

[doesn't need to be fixed, just couldn't help but mention it :)]